### ISSUE TITLE
Image create/resize errors

### ISSUE THREAD

**Issue by capnm** [godotengine/godot#21479]
**Godot version:**

49cf675ef460eb8baac7b5e598cd018f53bcec39

**OS/device including version:**

Linux

**Issue description:**

After the recent changes there are many such errors:

```ERROR: resize: Condition ' alloc->lock > 0 ' is true. returned: ERR_LOCKED
   At: core/dvector.h:541.
ERROR: create: Expected data size of 4096 bytes in Image::create(), got instead 8192 bytes.
   At: core/image.cpp:1454.
ERROR: resize: Condition ' alloc->lock > 0 ' is true. returned: ERR_LOCKED
   At: core/dvector.h:541.
```

```
(gdb) b core/image.cpp:1454
Breakpoint 3 at 0x555557788fdf: file core/image.cpp, line 1454.
(gdb) c
Continuing.
[New Thread 0x7fffe0530700 (LWP 30698)]
[Thread 0x7fffe0530700 (LWP 30698) exited]
[New Thread 0x7fffe0530700 (LWP 30704)]
[Thread 0x7fffe0530700 (LWP 30704) exited]
ERROR: resize: Condition ' alloc->lock > 0 ' is true. returned: ERR_LOCKED
   At: core/dvector.h:541.

Thread 1 "godot3" hit Breakpoint 3, Image::create () at core/image.cpp:1454
1454			ERR_FAIL_COND(p_data.size() != size);
(gdb) bt
#0  Image::create () at core/image.cpp:1454
#1  0x0000555557789b72 in Image::Image () at core/image.cpp:1845
#2  0x0000555555f6e5de in RasterizerStorageGLES3::texture_get_data () at drivers/gles3/rasterizer_storage_gles3.cpp:1100
#3  0x000055555733dd65 in VisualServerRaster::texture_get_data () at servers/visual/visual_server_raster.h:154
#4  0x000055555738be80 in CommandQueueMT::CommandRet2<VisualServer, Ref<Image> (VisualServer::*)(RID, int) const, RID, int, Ref<Image> >::call
    () at core/command_queue_mt.h:307
#5  0x00005555573680ea in CommandQueueMT::flush_one () at core/command_queue_mt.h:425
#6  CommandQueueMT::flush_all () at core/command_queue_mt.h:465
#7  VisualServerWrapMT::sync () at servers/visual/visual_server_wrap_mt.cpp:90
#8  0x0000555555784f21 in Main::iteration () at main/main.cpp:1799
#9  0x000055555577c6e1 in OS_X11::run () at platform/x11/os_x11.cpp:2736
#10 0x000055555576e07c in main () at platform/x11/godot_x11.cpp:55
(gdb) l
1449		int mm;
1450		int size = _get_dst_image_size(p_width, p_height, p_format, mm, p_use_mipmaps ? -1 : 0);
1451	
1452		if (size != p_data.size()) {
1453			ERR_EXPLAIN("Expected data size of " + itos(size) + " bytes in Image::create(), got instead " + itos(p_data.size()) + " bytes.");
1454			ERR_FAIL_COND(p_data.size() != size);
1455		}
1456	
1457		height = p_height;
1458		width = p_width;
(gdb) n
ERROR: create: Condition ' p_data.size() != size ' is true.
   At: core/image.cpp:1454.
1462	}
```

**Comment by akien-mga** [comment]
Confirmed, this is reproducible simply by opening a newly created project (with just icon.png).

**Comment by akien-mga** [comment]
It's a regression from #21467, CC @elasota.

**Comment by akien-mga** [comment]
Fixed by #21483.
